docs/db-migration-guide.ipynb

```diff
@@ -0,0 +1,237 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Obtain existing database credentials\n",
+    "\n",
+    "To obtain database credential execute the following commands for each database to be migrated and use those credentials in migration steps."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# replace DB_UNIT with desired unit name: `katib-db` or `kfp-db`\n",
+    "DB_UNIT= katib-db | kfp-db\n",
+    "\n",
+    "# obtain username and password of existing MariadDB database from DB relation\n",
+    "# in Charmed Kubeflow 1.6 and 1.7 username used is `root` and password is specified in `mysql` relation by\n",
+    "# 'root_password'\n",
+    "DB_MYSQL_RELATION_ID=$(juju show-unit $DB_UNIT/0 | yq '.[] | .relation-info | select(.[].endpoint == \"mysql\") | .[0] | .relation-id')\n",
+    "DB_USER=root\n",
+    "DB_PASSWORD=$(bash -c \"juju run --unit $DB_UNIT/0 'relation-get -r $DB_MYSQL_RELATION_ID - $DB_UNIT/0' | grep root_password\" | awk '{print $2}')\n",
+    "DB_IP=$(juju show-unit $DB_UNIT/0 | yq '.[] | .address')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Deploy MySQL databases\n",
+    "\n",
+    "Deploy new MySQL database and obtain credentials by executing the following commands:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# replace MYSQL_DB with desired application name: `katib-db-mysql` or `kfp-db-mysql`\n",
+    "MYSQL_DB= katib-db-mysql | kfp-db-mysql\n",
+    "\n",
+    "# deploy new MySQL database charm\n",
+    "juju deploy mysql-k8s $MYSQL_DB --channel 8.0/stable --trust\n",
+    "\n",
+    "# obtain username and password of new MySQL database from MySQL charm\n",
+    "MYSQL_DB_USER=$(juju run-action $MYSQL_DB/0 get-password --wait | yq '.[] | .results.username')\n",
+    "MYSQL_DB_PASSWORD=$(juju run-action $MYSQL_DB/0 get-password --wait | yq '.[] | .results.password')\n",
+    "MYSQL_DB_IP=$(juju show-unit $MYSQL_DB/0 | yq '.[] | .address')\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Katib DB migration\n",
+    "\n",
+    "Use credentials and information obtained in previous steps for `katib-db` to perform database migration by executing the following commands:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# ensure that there are no new connections are made and that dababase is not altered\n",
+    "# remove relation between katib-db-manager charm and katib-db charm\n",
+    "juju remove-relation katib-db-manager katib-db\n",
```

Comment:
Should we remove the relation after we have performed the database backup?

Reply:
The goal of relation removal is to prevent DB from being updated, i.e. no relation = not updates from other components. Will this actually do the trick?
Is there a better way to ensuring there is no communication with DB while we are doing backup?

Reply:
Then we expect a downtime, perhaps we should mention that in the guide.